pandas/core/resample.py

```diff
@@ -1271,8 +1272,11 @@ def _get_range_edges(first, last, offset, closed='left', base=0):
         if (is_day and day_nanos % offset.nanos == 0) or not is_day:
             return _adjust_dates_anchored(first, last, offset,
                                           closed=closed, base=base)
-
-    if not isinstance(offset, Tick):  # and first.time() != last.time():
+    # GH12351
```

Comment:
not what I meant. This is not a general fix at all. This is the same as before. The question is what kind of offset needs to do this kind of rollback / normalization.